### PR TITLE
docs: migrated `provider` docs snippets

### LINKED ISSUE #2951: Migrate snippets from `guide/provider`
- [ ] `pagination.test.ts`
- [ ] `provider.test.ts`
- [ ] `querying-the-chain.test.ts`

### PR DESCRIPTION
**Pull request by petertonysmith94**
- Closes #2951

# Summary

- Migrated `provider` docs snippets.
- I added the `PageInfo` type as an export from the `provider`.

# Checklist

- [X] All **changes** are **covered** by **tests** (or not applicable)
- [X] All **changes** are **documented** (or not applicable)
- [X] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [X] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

**Review by petertonysmith94 — COMMENTED** [review]
I'm getting the following [error](https://github.com/FuelLabs/fuels-ts/actions/runs/11501639826/job/32014691270?pr=3357#step:6:1881) in the CI, even though I haven't changed the documentation structure.

@FuelLabs/sdk-ts Has anyone else witnessed this?

**Comment by Torres-ssf** [comment]
> _**Coverage Report:**_
> 
> Lines	Branches	Functions	Statements
> 71.96%(-0.91%)	70.14%(-0.1%)	74.44%(-0.19%)	72.24%(-0.89%)
> Changed Files:

@petertonysmith94 I have never seen this one. 

Since you didn’t change any doc page structure or path, I suspect the issue might be related to a misconfiguration in the docs-hub runnable scripts.

**Comment by petertonysmith94** [comment]
> > _**Coverage Report:**_
> > Lines	Branches	Functions	Statements
> > 71.96%(-0.91%)	70.14%(-0.1%)	74.44%(-0.19%)	72.24%(-0.89%)
> > Changed Files:
> 
> @petertonysmith94 I have never seen this one.
> 
> Since you didn’t change any doc page structure or path, I suspect the issue might be related to a misconfiguration in the docs-hub runnable scripts.

@Torres-ssf it was a change to the GitHub action for the API docs migration. I've merged the following PR to [revert the changes](https://github.com/FuelLabs/github-actions/pull/46) (hopefully fixing this issue).

**Review by nedsalk — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

File: apps/docs-snippets2/src/provider/functionality-get-coins.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Provider, Wallet } from 'fuels';
+
+import { LOCAL_NETWORK_URL, WALLET_PVT_KEY } from '../env';
+
+const provider = await Provider.create(LOCAL_NETWORK_URL);
+const wallet = Wallet.fromPrivateKey(WALLET_PVT_KEY, provider);
```
Comment (maschad) [inline apps/docs-snippets2/src/provider/functionality-get-coins.ts:7]: If we want the snippet to be copy/pasteable shouldn't these imports be included in the `getCoins-1` region?

File: apps/docs-snippets2/src/provider/functionality-get-message-proof.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+const messageProofFromBlockId = await provider.getMessageProof(
+  result.id,
+  nonce,
+  latestBlock?.id
+);
+// #endregion getMessageProof-blockId
+
+// #region getMessageProof-blockHeight
+// Retrieves the message proof for the transaction ID and nonce using the block height
+const messageProofFromBlockHeight = await provider.getMessageProof(
```
Comment (maschad) [inline apps/docs-snippets2/src/provider/functionality-get-message-proof.ts:42]: Should this be a separate snippet?

File: apps/docs-snippets2/src/provider/functionality-get-transactions.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// #region getTransactions
+import { Provider } from 'fuels';
+
+import { LOCAL_NETWORK_URL } from '../env';
+
+const provider = await Provider.create(LOCAL_NETWORK_URL);
+
+const { transactions } = await provider.getTransactions();
+// #endregion getTransactions
+// #endregion getBlocks
```
Comment (maschad) [inline apps/docs-snippets2/src/provider/functionality-get-transactions.ts:10]:
```suggestion
```

**Review by maschad — COMMENTED** [review]
LGTM but left some comments

File: apps/docs-snippets2/src/provider/functionality-get-message-proof.ts (hunk 2 above)
Comment (petertonysmith94) [reply apps/docs-snippets2/src/provider/functionality-get-message-proof.ts]: [7a35699](https://github.com/FuelLabs/fuels-ts/pull/3357/commits/7a3569963ab403b40e35dda9ee03f5f06fbd210a)

File: apps/docs-snippets2/src/provider/functionality-get-transactions.ts (hunk 3 above)
Comment (petertonysmith94) [reply apps/docs-snippets2/src/provider/functionality-get-transactions.ts]: [967d76d](https://github.com/FuelLabs/fuels-ts/pull/3357/commits/967d76de77701408a356e9dea5a97cd686c23132)

File: apps/docs-snippets2/src/provider/functionality-get-coins.ts (hunk 1 above)
Comment (petertonysmith94) [reply apps/docs-snippets2/src/provider/functionality-get-coins.ts]: [16b1343](https://github.com/FuelLabs/fuels-ts/pull/3357/commits/16b1343715ac3718e060a748b98ea7030c0550ca)

**Review by Torres-ssf — APPROVED** [review]

**Review by maschad — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**